src/analysis/buildTypeWeights.ts

```diff
@@ -157,14 +154,42 @@ function parseObjectFields(
                     }
                 });
             }
+        } else if (isUnionType(fieldType)) {
+            // Users must query union types using inline fragments to resolve field specific to one of the types in the union
+            // however, if a type is shared by all types in the union it can be queried outside of the inline fragment
+            // any common fields should be added to fields on the union type itself in addition to the comprising types
+            result.fields[field] = {
+                resolveTo: fieldType.name.toLocaleLowerCase(),
+            };
```

Comment:
This action for an Union Type looks identical to the action for line 93 when checking for an Object, Interface or Enum. Is there any reason to put this in it's own `else if` statement?

Reply:
Good catch. I originally separated this out since I thought we needed extra functionality.